### DISCUSSION TITLE
bool.TryParse Improvement

### DISCUSSION THREAD

**Discussion by picoage** [dotnet/csharplang#7123] [Language Ideas] [answered]
Currently we use the following for bool.TryParse

```
  string value = "true";

            if (bool.TryParse(value, out bool flag))
            {
                if(flag)
                {
                    //Do Something if flag is true 
                }
            }
```

However, every time we need to check if the flag is true and do something. 

If we can have pattern matching, something similar to the following, will save us from writing addition line of code: 

 ```
    if (bool.TryParse(value, out bool flag?))
            {
                // To something if flag is true

                if (!flag)
                {
                    // If needed to something, if the flag is false 
                }
            }
```

Reason for proposing: Every time we are forced to check by adding addition if statement to see if the flag is true. 

Most of the time we only check for true and only in rare cases we check for false.

**Comment by HaloFour** [comment, marked as answer]
You do not need an additional `if` statement:

```cs
if (bool.TryParse(value, out bool flag) && flag)
{

}
```

**Reply by picoage** [reply]
@HaloFour , thanks for the feedback.
